freqtrade/freqtradebot.py

```diff
@@ -647,10 +648,38 @@ def handle_stoploss_on_exchange(self, trade: Trade) -> bool:
                 trade.sell_reason = SellType.STOPLOSS_ON_EXCHANGE.value
                 trade.update(order)
                 result = True
-            else:
-                result = False
+            elif self.config.get('trailing_stop', False):
+                # if trailing stoploss is enabled we check if stoploss value has changed
+                # in which case we cancel stoploss order and put another one with new
+                # value immediately
+                self.handle_trailing_stoploss_on_exchange(trade, order)
+
         return result
 
+    def handle_trailing_stoploss_on_exchange(self, trade: Trade, order):
+        """
+        Check to see if stoploss on exchange should be updated
+        in case of trailing stoploss on exchange
+        :param Trade: Corresponding Trade
+        :param order: Current on exchange stoploss order
+        :return: None
+        """
+
+        if trade.stop_loss > float(order['info']['stopPrice']):
+            # we check if the update is neccesary
+            update_beat = self.strategy.order_types['stoploss_on_exchange_interval']
```

Comment:
because the strategy-resolver replaces the whole order_types dict from either config, or strategy (or falls back to what is in interface.py), not single values.